From 80518005ca3ebe81c0a1cb0e821e47175231059f Mon Sep 17 00:00:00 2001
From: Sandor Yu <R01008@freescale.com>
Date: Thu, 20 Nov 2014 16:28:07 +0800
Subject: [PATCH 0335/1594] MLK-9779-04 dts: imx6sl: Add ov5640 camera to evk
 dts

commit b9eb439a210f3fc323b6f4c586594fdcd746dfb8 from
git://git.freescale.com/imx/linux-2.6-imx.git

- Add imx6sx-19x19-arm2-csi.dtb file
- Remove int-device v4l2 capture setting
- Add ov5640 camera endport setting
- Enable csi item.

Signed-off-by: Sandor Yu <R01008@freescale.com>
---
 arch/arm/boot/dts/Makefile                  |    1 +
 arch/arm/boot/dts/imx6sl-evk-csi.dts        |    6 -----
 arch/arm/boot/dts/imx6sl-evk.dts            |   32 +++++++++++++++++++++-----
 arch/arm/boot/dts/imx6sl.dtsi               |    6 ++++-
 arch/arm/boot/dts/imx6sx-19x19-arm2-csi.dts |   25 +++++++++++++++++++++
 5 files changed, 57 insertions(+), 13 deletions(-)
 create mode 100644 arch/arm/boot/dts/imx6sx-19x19-arm2-csi.dts

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index 619ceed..3548db5 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -336,6 +336,7 @@ dtb-$(CONFIG_SOC_IMX6SX) += \
 	imx6sx-17x17-arm2-gpmi-weim.dtb \
 	imx6sx-sdb.dtb \
 	imx6sx-19x19-arm2.dtb \
+	imx6sx-19x19-arm2-csi.dtb \
 	imx6sx-sdb-canfd.dtb \
 	imx6sx-sdb-emmc.dtb \
 	imx6sx-sdb-m4.dtb \
diff --git a/arch/arm/boot/dts/imx6sl-evk-csi.dts b/arch/arm/boot/dts/imx6sl-evk-csi.dts
index 795be40..56d824b 100644
--- a/arch/arm/boot/dts/imx6sl-evk-csi.dts
+++ b/arch/arm/boot/dts/imx6sl-evk-csi.dts
@@ -8,12 +8,6 @@
 
 #include "imx6sl-evk.dts"
 
-/ {
-	csi_v4l2_cap {
-		status = "okay";
-	};
-};
-
 &csi {
 	status = "okay";
 };
diff --git a/arch/arm/boot/dts/imx6sl-evk.dts b/arch/arm/boot/dts/imx6sl-evk.dts
index fbf002d..b1c97ba 100644
--- a/arch/arm/boot/dts/imx6sl-evk.dts
+++ b/arch/arm/boot/dts/imx6sl-evk.dts
@@ -40,11 +40,6 @@
 		default-brightness-level = <6>;
 	};
 
-	csi_v4l2_cap {
-		compatible = "fsl,imx6sl-csi-v4l2";
-		status = "disabled";
-	};
-
 	pxp_v4l2_out {
 		compatible = "fsl,imx6sl-pxp-v4l2";
 		status = "okay";
@@ -164,7 +159,11 @@
 };
 
 &csi {
-	status = "disabled";
+	port {
+		csi_ep: endpoint {
+			remote-endpoint = <&ov564x_ep>;
+		};
+	};
 };
 
 &cpu0 {
@@ -419,6 +418,27 @@
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_i2c3>;
 	status = "disabled";
+
+	ov564x: ov564x@3c {
+		compatible = "ovti,ov564x";
+		reg = <0x3c>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_csi_0>;
+		clocks = <&clks IMX6SL_CLK_CSI>;
+		clock-names = "csi_mclk";
+		AVDD-supply = <&vgen6_reg>;  /* 2.8v */
+		DVDD-supply = <&vgen2_reg>;  /* 1.5v*/
+		pwn-gpios = <&gpio1 25 1>;
+		rst-gpios = <&gpio1 26 0>;
+		csi_id = <0>;
+		mclk = <24000000>;
+		mclk_source = <0>;
+		port {
+			ov564x_ep: endpoint {
+				remote-endpoint = <&csi_ep>;
+			};
+		};
+	};
 };
 
 &iomuxc {
diff --git a/arch/arm/boot/dts/imx6sl.dtsi b/arch/arm/boot/dts/imx6sl.dtsi
index defe385..14aa433 100644
--- a/arch/arm/boot/dts/imx6sl.dtsi
+++ b/arch/arm/boot/dts/imx6sl.dtsi
@@ -683,9 +683,13 @@
 			};
 
 			csi: csi@020e4000 {
-				compatible = "fsl,imx6sl-csi";
+				compatible = "fsl,imx6s-csi";
 				reg = <0x020e4000 0x4000>;
 				interrupts = <0 7 IRQ_TYPE_LEVEL_HIGH>;
+				clocks = <&clks IMX6SL_CLK_DUMMY>,
+					<&clks IMX6SL_CLK_DUMMY>,
+					<&clks IMX6SL_CLK_DUMMY>;
+				clock-names = "disp-axi", "csi_mclk", "disp_dcic";
 				status = "disabled";
 			};
 
diff --git a/arch/arm/boot/dts/imx6sx-19x19-arm2-csi.dts b/arch/arm/boot/dts/imx6sx-19x19-arm2-csi.dts
new file mode 100644
index 0000000..e1b2c2d
--- /dev/null
+++ b/arch/arm/boot/dts/imx6sx-19x19-arm2-csi.dts
@@ -0,0 +1,25 @@
+/*
+ * Copyright (C) 2014 Freescale Semiconductor, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include "imx6sx-19x19-arm2.dts"
+
+&esai {
+	/* pin conflict with sai */
+	status = "disabled";
+};
+
+&sai1 {
+	status = "disabled";
+};
+
+&i2c2 {
+	ov564x: ov564x@3c {
+		status = "okay";
+	};
+};
+
-- 
1.7.5.4

